source/auth/auth.md

```diff
@@ -1647,8 +1647,8 @@ Use the following algorithm to authenticate a new connection:
 
 - Check if the the *Client Cache* has an access token.
   - If it does, cache the access token in the *Connection Cache* and perform a `One-Step` SASL conversation using the
-    access token in the *Client Cache*. If the server returns an error, invalidate that access token, sleep 100ms then
-    continue.
+    access token in the *Client Cache*. If the server returns a Authentication error (18), invalidate that access token,
+    sleep 100ms then continue.
```

Comment:
In a [prior discussion](https://mongodb.slack.com/archives/C04PGGEFCNM/p1708624000650649?thread_ts=1708109719.923909&cid=C04PGGEFCNM), I believe the plan was to remove the 100ms sleep, and leave it up to the callback.

The problem was "The 20 threads would not hit the callback all at once, but one after the other without a 100ms delay, which only applies to the first thread." - I don't recall the logic, but I think this means that the throttling would not be effective in certain common cases. (Also, there is no need to throttle AWS file-reading.)

Perhaps this is not the right PR to address this, but I think the DRIVERS-2836 ticket was the only one mentioned (so, if not here, could we create a ticket?), and I noticed a comment on the ticket related to the timeout as well.

Reply:
Updated